src/lib/radio/radio.ts

```diff
@@ -280,10 +300,26 @@ export class MdRadioButton implements OnInit {
   name: string;
 
   /** Used to set the 'aria-label' attribute on the underlying input element. */
-  @Input('aria-label') ariaLabel: string;
+  _ariaLabel: string;
+  @Input('aria-label')
+  get ariaLabel(): string {
+    return this._ariaLabel;
+  }
+  set ariaLabel(value: string) {
+    this._ariaLabel = value;
+    this._change.markForCheck();
```

Comment:
@jelbourn your understanding is correct, an `@Input()` property updated via a binding would mark the components for change detection. (it would not if it accessed directly, ie `myDirective.inputProperty = ...` in some component).

@tinayuangao 
> Then there's a bug for OnPush since the properties are not updated when their values changed.

I'm not sure to get what you mean here.

Only components are nodes in the change detection tree. Directives are checked by the CD together with their component and I expect that setting any of their input would have the same effect as setting a component input - I have to double check this case.